flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableInputFormat.java

```diff
@@ -88,17 +86,23 @@
 	private transient InputFormat mapredInputFormat;
 	private transient HiveTablePartition hiveTablePartition;
 
+	// indices of fields to be returned, with projection applied (if any)
+	// TODO: push projection into underlying input format that supports it
```

Comment:
I don't think underlying input format can support project push down. Any information or delete this comment?

Reply:
I haven't checked the code but I think columnar storage formats should be able to support projection pushdown. For example, these slides imply projection pushdown is supported in ORC and Parquet:
https://www.slideshare.net/oom65/orc-files?qid=a032419f-e6a3-4776-8c9c-03dd752f17fd&v=&b=&from_search=2
https://www.slideshare.net/julienledem/parquet-stratany-hadoopworld2013?qid=94dc3a84-39f0-456b-8766-f623e679a7ca&v=&b=&from_search=4

Reply:
I think orc/parquet is another story, we should use our implementation instead of hadoop `RecordReader`.
What I want to ask is: can `RecordReader` support project push down?
If it is a easy way, Can we support it now?

Reply:
> can RecordReader support project push down?

I guess it depends on the implementation. For example, Hive has its own [ORC input format](https://github.com/apache/hive/blob/rel/release-2.3.4/ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java). And it seems Hive pushes down the projection via [configurations](https://github.com/apache/hive/blob/rel/release-2.3.4/serde/src/java/org/apache/hadoop/hive/serde2/ColumnProjectionUtils.java#L52).

> Can we support it now?

I'd rather leave it to a separate task because it needs further investigation. Even though this PR doesn't reduce the amount of data to read, it avoids the costs of inspecting unused columns, which is good for queries selecting only a few columns from tables with lots of columns. And for input format that doesn't support projection push down (e.g. text), this is probably the best we can do.